shm/Ship.hpp

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <string>
+
+class Ship
+{
+public:
+    Ship() : id_(-1) {}
+    Ship(int capacity, int maxCrew, int speed, const std::string& name, size_t id)
+        : capacity_(capacity), maxCrew_(maxCrew), crew_(0), speed_(speed), name_(name), id_(id)
+    {
+    }
+    Ship(int maxCrew, int speed, size_t id) : Ship(0, maxCrew, speed, "", id) {}
+
+    void setName(const std::string& name) { name_ = name; }
+
+    Ship& operator-=(size_t num)
+    {
+        crew_ -= num;
+        return *this;
+    }
+    Ship& operator+=(size_t num)
+    {
+        crew_ += num;
+        return *this;
```

Comment:
How are you handling if you have space for more crew?

Reply:
+1